src/main/java/AudioPlayer.java

```diff
@@ -0,0 +1,240 @@
+     * during the playing, then the playing is restored from the same song, the execution of the song continues from the same timing.
+     * Otherwise if a new song is playing new song duration is set.
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public boolean executeSong() throws IOException, InterruptedException {
+        if (!wasSongPaused) {
+            songDurationLeft = songs.get(currentSongIndex).getTiming();
+        } else {
+            wasSongPaused = false;
+        }
+        for (; songDurationLeft > 0; songDurationLeft--) {
+            Thread.sleep(100);
+            if (applicaiton.isStateChanged()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    /**
+     * This method set the index, which points to the current song to the first one and changes the current state to PLAY
+     */
+    public void replay() {
+        currentSongIndex = FIRST_SONG_INDEX;
+    }
+
+    /**
+     * This method pauses the song until the "play" command is given in the console
+     */
+    public void pauseSong() throws IOException {
+        wasSongPaused = true;
+        while (!applicaiton.isStateChanged()) ;
+    }
+
+    /**
+     * This method switch to the previous song of the audio player, if the current song is the first one it switch to the last
+     */
+    public void prevSong() {
+        if ((currentSongIndex - 1) >= 0) {
+            currentSongIndex--;
+        } else {
+            currentSongIndex = songs.size() - 1;
+        }
+    }
+
+    /**
+     * This method switch to the next song of the audio player, if the current song is the last one it switch to the first
+     */
+    public void nextSong() {
+        if ((currentSongIndex + 1) < songs.size()) {
+            currentSongIndex++;
+        } else {
+            currentSongIndex = FIRST_SONG_INDEX;
+        }
+    }
+
+    /**
+     * This method is used to get specific information about the song which is the song's title, song's number in the list
+     * and the singer's name in string format
+     *
+     * @return String - this method return song's title, song's number and singer's name in a string format
+     */
+    public String getSongInfo() {
+        StringBuilder songInfo = new StringBuilder();
+        songInfo.append("\n\tCurrent song: ")
+                .append(songs.get(currentSongIndex).getTitle())
+                .append("\n\tNumber in the list: ")
+                .append(currentSongIndex + 1)
+                .append("\n\tSinger: ")
+                .append(songs.get(currentSongIndex).getSinger().getName());
+
+        return songInfo.toString();
+    }
+
+    /**
+     * This method plays the songs in the list, but in random order. If the application state is changed the shuffle is interrupted
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public void shuffle() throws IOException, InterruptedException {
```

Comment:
Here, you there is something tricky :) 
If `shuffle` is triggered, and I want `previous` song, then the result will be `currentIndex - 1`. 
You can use a flag `isShuffled`, but think how we can you it ;)